hardware_interface/src/components/component_lists_management.hpp

```diff
@@ -41,51 +40,38 @@ namespace components
   * list is is empty; return_type::OK otherwise.
   */
 inline return_type get_internal_values(
-  std::vector<double> & values, const std::vector<std::string> & queried_interfaces,
-  const std::vector<std::string> & int_interfaces, const std::vector<double> & int_values)
+  std::vector<double> & values, const std::vector<InterfaceInfo> & queried_interfaces,
```

Comment:
as commented above in `joint.cpp`

Reply:
not 100% sure what you mean. You mean, rename `int_interfaces` to `internal_interfaces`?

Reply:
I meant to keep `std::string` as the input type for the API simplicity.

It's not really that one need `min` and `max` information here. And we can simplify the calls of those functions from the component's implementation.

Reply:
and `internal_interfaces` is also way to go

Reply:
ok -- done